arch/src/aarch64/fdt.rs

```diff
@@ -159,6 +159,19 @@ pub fn write_fdt_to_memory(fdt_final: Vec<u8>, guest_mem: &GuestMemoryMmap) -> R
     Ok(())
 }
 
+pub fn num_cpus(dtb_path: &PathBuf) -> io::Result<u8> {
+
+    let mut file = File::open(dtb_path)?;
+    let pos = file.seek(SeekFrom::Current(0)).unwrap();
+    let metadata = file.metadata().unwrap();
+    let mut buffer = vec![0; metadata.len() as usize];
+    file.read_exact(&mut buffer).unwrap();
+    file.seek(SeekFrom::Start(pos)).unwrap();
+    let fdt = fdt_parser::Fdt::new(&buffer).unwrap();
```

Comment:
I didn't realize this already existed in the codebase... this should make it easy to extract the info we need...